### PR TITLE
Do not enable PERMISSIVE_HOLD when TAPPING_TERM exceeds 500ms

### PR DESCRIPTION
**Pull request by precondition**
## Description
Do not enable PERMISSIVE_HOLD-like behaviour when TAPPING_TERM ≥ 500ms. 

This is a poorly documented obscure "feature" that's entirely unnecessary and does more harm than good. When people are testing dual-role keys like `MT()`, they need to play around with the different tap-hold option and tweak their tapping term in order to limit misfires. A lot of times, people enable `IGNORE_MOD_TAP_INTERRUPT` because this is the option that is the most restrictive when it comes to selecting the hold action (desirable for a person using home row mods for example) but because they're not used to it, they get a lot of misfires. In which case, they're advised to increase the tapping term. 

However, some users crank it way above 500ms "just to be safe" and make it very hard to accidentally trigger a hold. That's when QMK "helpfully" enables permissive hold (despite the user not having enabled the setting in `config.h`) under their nose. Now, when testing, the user encounters *even more* misfires than before because permissive hold activates an additional way to trigger holds. This only serves to frustrate the users and push them away from exploiting the full capacities of the advanced tap-hold mechanics that QMK offers.

Finally, this PR is not a loss of functionality because one only needs to add `#define PERMISSIVE_HOLD` to their `config.h` to get the behaviour if they so desire. Furthermore, you can expect this PR to free up about 55 bytes in keymaps that don't enable `PERMISSIVE_HOLD`. (Everything counts on AVR)

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]
code change looks good.  But I'm not sure that this is for the best.

File: quantum/action_tapping.c [hunk 1]
```diff
@@ -164,15 +164,14 @@ bool process_tapping(keyrecord_t *keyp) {
                  * useful for long TAPPING_TERM but may prevent fast typing.
                  */
                 // clang-format off
-#    if defined(TAPPING_TERM_PER_KEY) || (TAPPING_TERM >= 500) || defined(PERMISSIVE_HOLD) || defined(PERMISSIVE_HOLD_PER_KEY) || (defined(AUTO_SHIFT_ENABLE) && defined(RETRO_SHIFT))
+#    if defined(PERMISSIVE_HOLD) || defined(PERMISSIVE_HOLD_PER_KEY) || (defined(AUTO_SHIFT_ENABLE) && defined(RETRO_SHIFT))
                 else if (
                     (
                         (
-                            GET_TAPPING_TERM(tapping_keycode, &tapping_key) >= 500
```
Comment (KarlK90) [inline quantum/action_tapping.c:171]: Same here.

**Comment by KarlK90** [comment]
> Can I also remove these (would-be-)outdated comments?

Yes, please remove them.

**Comment by KarlK90** [comment]
One more request:

```
❯ rg "TAPPING_TERM [5-9]\d\d"
users/wanleg/config.h
14:#define TAPPING_TERM 700

keyboards/1k/keymaps/tap_dance/config.h
6:#define TAPPING_TERM 500

keyboards/idobao/id67/keymaps/thewerther/config.h
20:#define TAPPING_TERM 500

keyboards/handwired/onekey/config.h
44:#define TAPPING_TERM 500

keyboards/handwired/ergocheap/config.h
49:#define TAPPING_TERM 500

keyboards/handwired/uthol/rev3/config.h
59:#define TAPPING_TERM 500

keyboards/xiudi/xd002/keymaps/tap_dance/config.h
3:#define TAPPING_TERM 500

keyboards/1upkeyboards/sweet16/keymaps/ridingintraffic/config.h
5:#define TAPPING_TERM 500

keyboards/keebio/quefrency/keymaps/bjohnson/config.h
29:#define TAPPING_TERM 500

keyboards/thevankeyboards/minivan/keymaps/halvves/config.h
3:#define TAPPING_TERM 505

keyboards/thevankeyboards/minivan/keymaps/belak/config.h
6:#define TAPPING_TERM 500

keyboards/flehrad/bigswitch/keymaps/333fred/config.h
26:#define TAPPING_TERM 500

keyboards/zfrontier/big_switch/config.h
53:#define TAPPING_TERM 500

layouts/community/66_ansi/xyverz/config.h
3:#define TAPPING_TERM 600 // ms
```

These configs should get `#define PERMISSIVE_HOLD` as they expected (?) or at least use this behavior.

**Comment by filterpaper** [comment]
Do we typically have to manage user config/keymaps for breaking changes?

**Comment by KarlK90** [comment]
> Do we typically have to manage user config/keymaps for breaking changes?

I would say that QMK should not break user space if it is possible. So if there is a trivial backwards compatible conversion it should be done in the PR.

**Comment by precondition** [comment]
> Yes, please remove them. This doesn't need confirmation from the owners as it just rectifies the situation.
Solved in https://github.com/qmk/qmk_firmware/pull/15674/commits/9167f41ca340a33edf2c13d694893dcb9810dd52

> These configs should get #define PERMISSIVE_HOLD as they expected (?) or at least use this behavior.

Solved in https://github.com/qmk/qmk_firmware/pull/15674/commits/f3e9fa777bee464f725904cb71b9c2b8aeac9174

File: quantum/action_tapping.c [hunk 2]
```diff
@@ -164,15 +164,14 @@ bool process_tapping(keyrecord_t *keyp) {
                  * useful for long TAPPING_TERM but may prevent fast typing.
                  */
                 // clang-format off
-#    if defined(TAPPING_TERM_PER_KEY) || (TAPPING_TERM >= 500) || defined(PERMISSIVE_HOLD) || defined(PERMISSIVE_HOLD_PER_KEY) || (defined(AUTO_SHIFT_ENABLE) && defined(RETRO_SHIFT))
```
[the remark this answers is not in the document]
Comment (KarlK90) [reply quantum/action_tapping.c:167]: Ah yes, you are totally right.

**Review by KarlK90 — APPROVED** [review]
Thanks for quickly implementing the changes.

File: quantum/action_tapping.c [hunk 3]
```diff
@@ -164,15 +164,14 @@ bool process_tapping(keyrecord_t *keyp) {
...
                     (
                         (
-                            GET_TAPPING_TERM(tapping_keycode, &tapping_key) >= 500
 #        ifdef PERMISSIVE_HOLD_PER_KEY
-                            || get_permissive_hold(tapping_keycode, &tapping_key)
+                            get_permissive_hold(tapping_keycode, &tapping_key)
 #        elif defined(PERMISSIVE_HOLD)
-                            || true
+                            true
 #        endif
```
Comment (KarlK90) [inline quantum/action_tapping.c:175]: That was a bit rushed, the unit-tests are failing now with:

```
Making test caps_word_autoshift                                                                                [ERRORS]
quantum/action_tapping.c: In function ‘process_tapping’:
quantum/action_tapping.c:176:25: error: expected expression before ‘)’ token
  176 |                         ) && IS_RELEASED(event) && waiting_buffer_typed(event)
      |                         ^
 | 
 | 
 | 
```

I guess we need a false at this position for all other cases?

```suggestion
                            true
#        else
                            false                     
#        endif
```

**Review by KarlK90 — CHANGES_REQUESTED** [review]

File: quantum/action_tapping.c (hunk 3 above)
Comment (KarlK90) [reply quantum/action_tapping.c]: Thanks!

**Review by KarlK90 — APPROVED** [review]
All test are green now. This is good to go.

**Comment by precondition** [comment]
It also passes all the new tap hold configuration unit tests I had added in https://github.com/qmk/qmk_firmware/pull/15741 so that's extra re-insurance

**Pull request merged**